File: app/src/main/java/com/kamontat/checkidnumber/model/Pool.java
package com.kamontat.checkidnumber.model;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.*;

/**
 * @author kamontat
 * @version 1.0
 * @since Thu 11/May/2017 - 9:20 PM
 */
public class Pool extends ArrayAdapter<IDNumber> {
	protected int size;
	protected ArrayList<IDNumber> idNumbers;
	
	public Pool(@NonNull Context context) {
		super(context, android.R.layout.two_line_list_item, android.R.id.text1);
		this.idNumbers = new ArrayList<>();
	}
	
	@Override
	public void add(@Nullable IDNumber object) {
		super.add(object);
		idNumbers.add(object);
	}
	
	@Override
	public void remove(@Nullable IDNumber object) {
		super.remove(object);
		idNumbers.remove(object);
	}
	
	@Override
	public void clear() {
		super.clear();
		idNumbers.clear();
	}
	
	@NonNull
	@Override
	public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
		if (convertView == null)
			convertView = LayoutInflater.from(getContext()).inflate(android.R.layout.two_line_list_item, parent, false);
		
		TextView textView1 = (TextView) convertView.findViewById(android.R.id.text1);
		TextView textView2 = (TextView) convertView.findViewById(android.R.id.text2);
		
		IDNumber idNumber = getItem(position);
		if (idNumber == null) return super.getView(position, convertView, parent);
		
		textView1.setText(idNumber.getId());
		
		textView2.setText(String.format(Locale.ENGLISH, "Status: %s", idNumber.getStatus().toString()));
		textView2.setTextColor(idNumber.getStatus().getColor(getContext().getResources()));
		
		return super.getView(position, convertView, parent);
	}
	
	public IDNumber[] getIDNumbers() {
		return idNumbers.toArray(new IDNumber[idNumbers.size()]);
	}
}

File: app/src/main/java/com/kamontat/checkidnumber/model/Pool.kt
package com.kamontat.checkidnumber.model

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import java.io.Serializable

import java.util.*

/**
 * @author kamontat
 * *
 * @version 1.0
 * *
 * @since Thu 11/May/2017 - 9:20 PM
 */
class Pool(context: Context) : ArrayAdapter<IDNumber>(context, android.R.layout.two_line_list_item, android.R.id.text1), Serializable {
    private var idNumbers: ArrayList<IDNumber> = ArrayList<IDNumber>()

    override fun add(`object`: IDNumber?) {
        super.add(`object`)
        `object`?.let { idNumbers.add(it) }
    }

    override fun remove(`object`: IDNumber?) {
        super.remove(`object`)
        idNumbers.remove(`object`)
    }

    override fun clear() {
        super.clear()
        idNumbers.clear()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null)
            view = LayoutInflater.from(context).inflate(android.R.layout.two_line_list_item, parent, false)

        val textView1 = view!!.findViewById(android.R.id.text1) as TextView
        val textView2 = view.findViewById(android.R.id.text2) as TextView

        val idNumber = getItem(position) ?: return super.getView(position, view, parent)

        textView1.text = idNumber.getId()

        textView2.text = String.format(Locale.ENGLISH, "Status: %s", idNumber.status.toString())
        textView2.setTextColor(idNumber.status.getColor(context.resources))

        return super.getView(position, view, parent)
    }

    fun getIDNumbers(): Array<IDNumber> {
        return idNumbers.toTypedArray()
    }
}
